assets/src/components/Sidebar/LegacyThemeSettings.js

```diff
@@ -30,108 +44,111 @@ const themeOptions = [
 
 ];
 
-const ThemeSelect = withPostMeta( SelectControl );
-
-const SelectWithDefaultLabel = compose(
-  fromThemeOptions,
-  withPostMeta,
-  withDefaultLabel,
-)( SelectControl );
-
-const Select = compose(
-  fromThemeOptions,
-  withPostMeta,
-)( SelectControl );
-
-const Radio = compose(
-  fromThemeOptions,
-  withPostMeta,
-)( RadioControl );
-
-const ColorPalette = compose(
-  fromThemeOptions,
-  withPostMeta,
-)( ColorPaletteControl );
-
 export const LegacyThemeSettings = props => {
   const {
     handleThemeSwitch,
     theme,
   } = props;
 
+  const meta = useSelect(select => select('core/editor').getEditedPostAttribute('meta'),[]);
+  const { editPost } = useDispatch('core/editor');
+
+  useEffect(() => {
+    handleThemeSwitch('theme', meta.theme, meta)
+  }, [meta.theme]);
+
+  // resolveField is a cheap function so it's not a problem to call it twice.
+  const getValue = fieldId => meta[fieldId] || resolveField(theme, fieldId, meta)?.default;
+  const getOptions = fieldId => resolveField(theme, fieldId, meta)?.options || [];
+
+  const updateValueAndDependencies = fieldId => value => {
+    const updatedDeps = getDependencyUpdates(theme, fieldId, value, meta);
+    editPost({ meta: {[fieldId]: value, ...updatedDeps} });
+  }
+
   return <>
     <div className="components-panel__body is-opened">
-      <ThemeSelect
-        metaKey='theme'
+      <SelectControl
+        options={themeOptions}
         label={ __( 'Theme', 'planet4-blocks-backend' ) }
-        options={ themeOptions }
-        getNewMeta={ handleThemeSwitch }
+        value={ meta.theme }
+        onChange={ value => {
+          editPost({ meta: { theme: value } });
+        } }
       />
     </div>
     <PanelBody
       title={ __( "Navigation", 'planet4-blocks-backend' ) }
       initialOpen={ true }
     >
-      <Radio
-        metaKey='campaign_nav_type'
-        theme={ theme }
+      <RadioControl
+        selected={getValue(LEGACY_FIELDS.navigationType)}
```

Comment:
Even though the new code is more redundant, you can now easily read the following:
- Which component is actually used.
- Which props are set on that component (was hidden inside of 2 different HOCs previously)
- What happens when you change it.